README.md

```diff
@@ -476,6 +509,79 @@ Client client = new Client();
 client.close();
 ```
 
+#### Intercepting Requests & Responses from a Client
+The request classes have [hooks for intercepting the request and the response](#request--response-interception)
+which the `Client` class leverages to provide an API for registering a chain of
+interceptors that will be applied to all requests and subsequent responses
+created by the client.
+
+```dart
+class HeaderInterceptor extends HttpInterceptor {
+  @override
+  Future<RequestPayload> interceptRequest(RequestPayload payload) async {
+    payload.request.headers['x-foo'] = 'bar';
+    return payload;
+  }
+}
+
+class QueryParamInterceptor extends HttpInterceptor {
+  @override
+  Future<RequestPayload> interceptRequest(RequestPayload payload) async {
+    payload.request.updateQuery({'baz': 'bar'});
+    return payload;
+  }
+}
+
+main() {
+  Client client = new Client()
+    ..addInterceptor(new HeaderInterceptor())
+    ..addInterceptor(new QueryParamInterceptor());
+
+  // The client will create a request interceptor that chains together the logic
+  // from both of the interceptors registered above. This will be set on the
+  // request, meaning that this request will have an `x-baz: bar` header and a
+  // `baz=bar` query parameter.
+  client.newRequest().get(uri: Uri.parse('...'));
+```
+
+Obviously these examples are contrived, but this pattern enables some powerful
+functionality. Consider the following interceptors as possibilities:
+
+- **Analytics:** records types of requests, request duration, request
```

Comment:
Sounds good! But in all seriousness, I think it would be cool/make sense to create a repo with useful interceptors like this. Or maybe those just live in a separate entry point in this repo.